packages/plugin-ext/src/main/browser/notebooks/renderers/cell-output-webview.tsx

```diff
@@ -354,21 +354,27 @@ export class CellOutputWebviewImpl implements CellOutputWebview, Disposable {
             this.webviewWidget.show();
         }
 
+        const visibleCells = this.notebook.getVisibleCells();
+
         const updateOutputMessage: OutputChangedMessage = {
             type: 'outputChanged',
-            changes: updates.map(update => ({
-                cellHandle: update.cellHandle,
-                newOutputs: update.newOutputs.map(output => ({
-                    id: output.outputId,
-                    items: output.outputs.map(item => ({ mime: item.mime, data: item.data.buffer })),
-                    metadata: output.metadata
-                })),
-                start: update.start,
-                deleteCount: update.deleteCount
-            }))
+            changes: updates
+                .filter(update => visibleCells.some(cell => cell.handle === update.cellHandle))
```

Comment:
Suggestion: This leads to `n * m` runtime behavior. Can we build something this instead and use `has`?

```ts
const visibleCellHandles = new Set(this.notebook.getVisibleCells().map(cell => cell.handle));
```

Reply:
not sure if thats really that worth it, since in 90% of cases this will probably only be a single update. But i changed it. Your probably right that it could improve performance for a few cases